src/cloudai/cli/handlers.py

```diff
@@ -83,6 +84,63 @@ def handle_install_and_uninstall(args: argparse.Namespace) -> int:
     return rc
 
 
+def is_dse_job(cmd_args):
+    """
+    Recursively check if any value in cmd_args is a list.
+
+    Args:
+        cmd_args (dict): The command arguments to check.
+
+    Returns:
+        bool: True if any value is a list, False otherwise.
+    """
+    if isinstance(cmd_args, dict):
+        for _key, value in cmd_args.items():
+            if isinstance(value, list) or (isinstance(value, dict) and is_dse_job(value)):
+                return True
+    return False
+
+
+def handle_dse_job(tr, system, test_scenario, args):
+    agent = GridSearchAgent(tr)
+    env = CloudAIGymEnv(test_run=tr, system=system, test_scenario=test_scenario)
+    agent.configure(env.action_space)
+
+    for dse_iteration, action in enumerate(agent.get_all_combinations(), start=1):
```

Comment:
This will work for grid search only, right? For smarter agents we are not able to get list of all combinations upfront.